### DISCUSSION TITLE
How does the sorting work?

### DISCUSSION THREAD

**Discussion by Etkiny** [Budibase/budibase#5301] [Help]
Hello,

It is a wonderful project, thanks for you effort!

There is  a problem with ordering the colomns. If the type is integer, It does not order properly. It is okay with varchar. I am having an query and I have the result of the query into a table. When I try to order by clicking on the colon name, there is a problem with integers. 

How does colomn ordering (asc/desc) work?

Thank you, guys!

**Comment by melohagan** [comment]
Hello @Etkiny 

Thank you for your kind words!

For the column sorting when the datasource is from a query, you want to make sure to have the desired data-type selected in the query schema:
![Screenshot 2022-04-08 at 13 55 47](https://user-images.githubusercontent.com/101575380/162439865-647172bd-043b-4b2b-8891-677475db2f2e.png)

If you wanted to sort by the text value instead (e.g. 1, 11, 2) then set the value type to *Text*. By default it will be text.

Does that answer your question, or is there anything else I can help with?

**Reply by Etkiny** [reply]
Hello @melohagan,

Thanks for your reply. This is actually the right answer of my question,tbank you. Eventhough it works on preview, It does not work when I play the real app to test it. It goes something like 

> 9,9,9,8,7,7,6,5,5,4,3,3,21, 2,2,17,15,14,12,1,0

**Reply by melohagan** [reply]
Make sure to click Save Query after you update the value data type to number. If you click the design tab and then back to the data tab  and look at the query, it should have kept the value type as number if it saved correctly. 

I have raised a bug here to improve the preview tab display: https://github.com/Budibase/budibase/issues/5347

**Reply by Etkiny** [reply]
@melohagan hello again, 
It still does not work :( I saved the query, but still wrong sorting. By the way, the preview tab works fine after I saved the query. However, after the preview I go back to schema and see that my changes dissappeared and all fields go back to "text".

**Reply by melohagan** [reply]
The schema can be a little funky. The order is Run Query, then update the schema data type, then Save Query. 
If you run query after saving, it will reset your custom schema data types.
